library/src/scala/compiletime/package.scala

```diff
@@ -68,6 +77,20 @@ package object compiletime {
     case t: T => t
   }
 
+  /** Given a tuple T, summons each of its member types and returns them in
+   *  a List.
+   *
+   *  @tparam T the tuple containing the types of the values to be summoned
+   *  @return the given values typed as elements of the tuple
+   */
+  inline def summonAll[T <: Tuple]: T =
+    val res =
+      inline erasedValue[T] match
+        case _: EmptyTuple => EmptyTuple
+        case _: (t *: ts) => summonInline[t] *: summonAll[ts]
+      end match
+    res.asInstanceOf[T]
```

Comment:
Same unsoundness issue as above.

Reply:
One way to fix it would be with a transparent inline but we have an issue that prevents the recursion from working properly.

Reply:
The other option is to use a match type.

```scala
inline def summonAll[T <: Tuple]: Tuple.Map[T, Identity] = ....
```

This way the tuple type is reconstructed but no singleton is kept.